### ISSUE TITLE
PHP 7.4 EoL

### ISSUE THREAD

**Issue by haderlump22** [nuxsmin/sysPass#1874]
**sysPass Version**
3.2.11

**Describe the question**
what is the plan for the expiring php version 7.4

**Screenshots**

**Platform (please complete the following information):**
 - OS: in this case, not important
 - OS Version: in this case, not important
 - Browser: in this case, not important

**Additional context**
No

**Comment by EchedelleLR** [comment]
Related: #1849

**Comment by ktoczyski** [comment]
Hi,

Is there any plans to support PHP 8+ in the near future? I use FreeBSD and now PHP is only available 8.0, 8.1 (default) and 8.2, other version was dropped.

Now I can't use sysPass because I received this info:

"Required PHP version >= 7.3 <= 7.4"

**Comment by vmario89** [comment]
getting the same problem in the future. i guess we simply could modify some configs to allow higher php version. but for a secure system it should be checked against all changed to ensure everything is still working correctly. sadly i am not a PHP dev :-(

**Comment by vmario89** [comment]
at least its not just a simple modification. We get more serious errors when chaning to php8.X like
`mod_fcgid: stderr: PHP Parse error:  syntax error, unexpected token "match" in /var/www/vhosts/<redacted>/vendor/phpunit/phpunit-mock-objects/src/Builder/ParametersMatch.php on line 17, referer: https://<redacted>/`

**Comment by nbit-ag** [comment]
same... we are trying to go to php version 8.2 but unfortunately without success... we get quite a lot of dependency errors.... :/

**Comment by Arash9703** [comment]
Same here , it doesn't work when the version of php is over 7.4 ! 
I hope this change is in future planning of syspass.

**Comment by Xiokar** [comment]
If you want to use sysPass with PHP 8.x then follow those steps : 

1. remove composer.lock and vendor directory
2. edit composer.json by replacing version of phpunit/phpunit by **"^9.3"** AND the package **"phpunit/dbunit": "^3"** by **"misantron/dbunit": "^5"**
![image](https://github.com/nuxsmin/sysPass/assets/696056/0bee5732-a8d7-495f-b3ae-05e0b2acd9dc)
3. save your composer.json
4. launch "composer install"
5. edit file lib/SP/Util/Checks.php (line 52) to change the max PHP version 7.4 to 9.0
![image](https://github.com/nuxsmin/sysPass/assets/696056/7b8cba20-304f-4897-89d0-057b0c6443e8)
6. save your file
7. Done, syspass is now fully working on PHP 8.x

**Comment by ktoczyski** [comment]
> If you want to use sysPass with PHP 8.x then follow those steps :
> 
> 1. remove composer.lock and vendor directory
> 2. edit composer.json by replacing version of phpunit/phpunit by **"^9.3"** AND the package **"phpunit/dbunit": "^3"** by **"misantron/dbunit": "^5"**

Hi,

I changed this in my composer file but I received this errors:

```
  Problem 1
    - Root composer.json requires php ~7.3 || ~7.4 but your php version (8.1.20) does not satisfy that requirement.
  Problem 2
    - Root composer.json requires doctrine/common ~v2.7.3 -> satisfiable by doctrine/common[v2.7.3].
    - doctrine/common v2.7.3 requires php ~5.6|~7.0 -> your php version (8.1.20) does not satisfy that requirement.
  Problem 3
    - fzaninotto/faker[v1.8.0, ..., v1.9.2] require php ^5.3.3 || ^7.0 -> your php version (8.1.20) does not satisfy that requirement.
    - Root composer.json requires fzaninotto/faker ^v1.8 -> satisfiable by fzaninotto/faker[v1.8.0, v1.9.0, v1.9.1, v1.9.2].
```

Your composer.json file is from SysPass repo?

**Comment by eduardomozart** [comment]
Thank you @Xiokar, I was able to follow your steps and make sysPass work on PHP 8.2.
@ktoczyski I only followed @Xiokar steps and added plugin-Authenticator to it. You'll still need to apply the patches https://github.com/sysPass/plugin-Authenticator/pull/36 and https://github.com/sysPass/plugin-Authenticator/pull/48 to make the Authenticator plugin works as expected.
[sysPass-3.2.11+vendor+plugin-Authenticator-2.2.2+PHP8.zip](https://github.com/nuxsmin/sysPass/files/14407774/sysPass-3.2.11%2Bvendor%2Bplugin-Authenticator-2.2.2%2BPHP8.zip)

**Comment by Xiokar** [comment]
@ktoczyski 
> Your composer.json file is from SysPass repo?

Yes, I had tested it 3 times on fresh installations of syspass before publishing here.
Did you followed the entire step ? Removing .lock file + vendor directory before other steps.

@eduardomozart 
You're welcome

**Comment by vmario89** [comment]
i can confirm that the solution from @Xiokar is working properly. Thank you for investigating.

i think it still will make sense to find a new sucessor as sysPass project is dead now for largely over 2 years. There is also no known active fork and nobody with PHP skills adopted or continued that work :/

**Comment by nuxsmin** [comment]
Hey guys, sysPass is alive and it will be soon updated to support PHP >=
8.2 among other useful features like CLI, dynamic config using environment
variables, full unit testing and performance and code improvements.

Sorry it's being a long and hard work, but it will provide a better and
more reliable base to implement new features.

Please check out the open PR about it ;)

**Comment by 4170R** [comment]
Are there any updates on this? Is there a planned publication date?

**Comment by patrice-go-dsi** [comment]
you are  the one @Xiokar !

**Comment by zezar98** [comment]
Thanks a lot @Xiokar , you are the man!!!

With you directions I was able to install syspass with php 8.2

**Comment by Xiokar** [comment]
Excellent, I'm glad to hear that this procedure is helping you all!
